### DISCUSSION TITLE
ui suggestions via figma

### DISCUSSION THREAD

**Discussion by reitenth** [IgnisDa/ryot#451] [Ideas]
im looking at the codebase and i wouldnt be able to help all the much on the frontend, so i was wondering if u would take some suggestions via figma? ive never used it but the jellyfin team is using to overhaul the ui and would like to contribute, specifically the frontend, if possible. here is a small peojevt i have worked on to show some design decisions in anlist. https://userstyles.world/style/10552/anilist-redesigned

**Comment by IgnisDa** [comment]
@reitenth Thanks for your interest. The frontend code is pretty straightforward, and I would prefer you contribute directly there. Figma is nice, but I don't really use it for designing (or any other tool).

What I suggest you to do is come up with whatever design you want, we discuss it here, and then you implement it slowly. No time better than now to learn something new. I can provide you instructions to generate mock data etc. if you need.

**Comment by reitenth** [comment]
okay it looks like the frontend is all typescript? ill will start learning more about it.

**Comment by reitenth** [comment]
could u help me get this up and running on vscode for windows 11. i followed some of this https://ignisda.github.io/ryot/architecture.html
but really am just unsure how to get debug and preview of changes. cargo build leads to alot of errors if thats even what im supposed to do.

**Reply by IgnisDa** [reply]
Please post the errors you're getting. We can go through them one by one.

**Reply by reitenth** [reply]
ok i got devcontainer working. i need to know how to open it in webpage to see changes 
![image](https://github.com/IgnisDa/ryot/assets/94771708/3b6cd7eb-cd18-46f6-b43e-e28219b17b73)

**Reply by IgnisDa** [reply]
You need to start both the servers using `mprocs`.

**Reply by reitenth** [reply]
![image](https://github.com/IgnisDa/ryot/assets/94771708/8f43635c-91ce-497b-9e9a-a3437d18712c)
cant seem to get moon installed, otherwise everything looks good

**Reply by reitenth** [reply]
![image](https://github.com/IgnisDa/ryot/assets/94771708/f0e2a010-3ed2-43c3-ab41-c0e5e3f685f5)

**Reply by IgnisDa** [reply]
You need to install moon manually. Go to the website for moonrepo.

**Reply by reitenth** [reply]
i cant get it installed on the dev container, i can get it installed on windows fine
![image](https://github.com/IgnisDa/ryot/assets/94771708/2b6302f5-08c6-4fca-a865-c09cd41c44b6)

**Reply by IgnisDa** [reply]
Install it globally. If it still does not work, join moon's discord (link on their website) and I'll help you there.

**Reply by reitenth** [reply]
i got it installed. im getting this error trying to run the frontend now
![image](https://github.com/IgnisDa/ryot/assets/94771708/de779d24-0ab0-4c1f-9c11-3e39f21df381)

**Reply by IgnisDa** [reply]
The file has been changed probably? Check all the package.json files and confirm that nothing has been changed. You can use git for this.

**Comment by reitenth** [comment]
got it running takes me to this page? is this correct not sure login credentials
![image](https://github.com/IgnisDa/ryot/assets/94771708/a6cd1225-e3e7-47d2-8f3f-64fdd98ab86d)
![image](https://github.com/IgnisDa/ryot/assets/94771708/33d1ead7-df86-40e5-8631-c91abb29b7f5)

**Reply by IgnisDa** [reply]
No this is the s3 storage. 

- backend: `http://localhost:8000`
- frontend: `http://localhost:3000` (this is probably what you want)
- s3: `http://localhost:9001` (credentials are username: `minioadmin`, password: `minioadmin`)

**Reply by reitenth** [reply]
localhost 3000 is leading to problem loading page. everything is running accoring to mprocs
![image](https://github.com/IgnisDa/ryot/assets/94771708/5ca5a6ef-a4aa-4190-a567-455cb25bde0c)

**Reply by IgnisDa** [reply]
I do not use vscode, but I have noticed that mprocs does not work inside vscode.

You can run `cargo watch -qx 'run --package ryot'` in one terminal and `moon run frontend:dev` in another if you want to use vscode.

**Reply by reitenth** [reply]
![image](https://github.com/IgnisDa/ryot/assets/94771708/ab7a4d35-48fe-4943-ad6d-b2be98a23a8b)
![image](https://github.com/IgnisDa/ryot/assets/94771708/0f72c344-464d-4c1d-a9c5-6557d9e806f8)
it seems like we are close. localhost still doesnt show anything however.

**Reply by IgnisDa** [reply]
Have you copied the config file in https://ignisda.github.io/ryot/architecture.html#environment? Specifically this line: `    "config_dump_path": "/tmp/ryot.json",`

**Reply by reitenth** [reply]
![image](https://github.com/IgnisDa/ryot/assets/94771708/ffbcee09-08f5-42e1-bf59-8bd8465a0e40)
![image](https://github.com/IgnisDa/ryot/assets/94771708/7ad74f1d-ff58-4c8f-9e7a-32bc034045a6)
![image](https://github.com/IgnisDa/ryot/assets/94771708/238790aa-cf33-485d-86c0-4beb93bc003f)

added, mprocs says frontend down now

**Reply by reitenth** [reply]
![image](https://github.com/IgnisDa/ryot/assets/94771708/3c7d40c3-377f-4e1d-8b6d-aff210e04613)

**Reply by IgnisDa** [reply]
Nope, you have the servers running, don't use mprocs. I use mprocs since I use vim. You can just use the commands I wrote above if you;re using VScode.

**Reply by IgnisDa** [reply]
Only you have a very old version of the code checked out. Clone the latest main branch.

**Reply by reitenth** [reply]
that seems to got it working! only issue rn is the login button doesnt do anything,
![image](https://github.com/IgnisDa/ryot/assets/94771708/e7d82e75-136c-4bf7-8763-4e5ee91cba2b)

**Reply by IgnisDa** [reply]
Open the network tab and then click the button. You will see a request to the graphql endpoint. Show me the response. Also the logs in the browser console if the cookie is being rejected.

**Reply by reitenth** [reply]
![image](https://github.com/IgnisDa/ryot/assets/94771708/8c38aa3d-b6b2-47b8-8603-39a9495f28c5)
![image](https://github.com/IgnisDa/ryot/assets/94771708/342c7794-c5e1-448a-9ab4-c84a32ab472b)

404 error, page could not be found. nothing on the logs tab of browser console. network tab says no cookies for this request

**Reply by IgnisDa** [reply]
You have not copied the env file required for the front-end. Please take a look at the architecture docs.

**Comment by reitenth** [comment]
i added 
          NEXT_PUBLIC_BASE_URL="http://localhost:8000" to /apps/frontend/.env
and the same thing to my settings.json in vscode and still not getting past login.
![image](https://github.com/IgnisDa/ryot/assets/94771708/a2e54ef9-c1d1-4a62-9879-84292de8fce2)

**Comment by IgnisDa** [comment]
Your requests are going to the wrong port. Maybe try restarting the devcontainer?

**Comment by reitenth** [comment]
restarted twice, doesnt do anything. here are some errors the network tab is showing.
![image](https://github.com/IgnisDa/ryot/assets/94771708/4682ff79-c04d-4939-8303-754b4606bce2)
![image](https://github.com/IgnisDa/ryot/assets/94771708/6efe0924-77d8-43e4-9587-a43ddc15db56)

**Reply by IgnisDa** [reply]
Your backend server is not running. There are errors. Fix them first.

**Reply by reitenth** [reply]
im running. these are the errors.

cargo watch -qx 'run --package ryot'
![image](https://github.com/IgnisDa/ryot/assets/94771708/dd16c40b-f3f5-45fd-9ef8-fba80b8435c8)

**Reply by IgnisDa** [reply]
You did not run `moon run frontend:build`.

**Reply by reitenth** [reply]
i did in the right panel. is it supposed to be in one? 
![image](https://github.com/IgnisDa/ryot/assets/94771708/4993a14c-6aea-4c65-90e2-2972732af58d)

**Reply by IgnisDa** [reply]
No read the docs again. It says you need to build once before dev.
